great_expectations/core/__init__.py

```diff
@@ -900,156 +1192,142 @@ def append_expectation(self, expectation_config):
         """
         self.expectations.append(expectation_config)
 
-    def find_expectation_indexes(
-        self, expectation_type=None, column=None, expectation_kwargs=None
-    ):
-        """Find matching expectations and return their indexes.
-        Args:
-            expectation_type=None                : The name of the expectation type to be matched.
-            column=None                          : The name of the column to be matched.
-            expectation_kwargs=None              : A dictionary of kwargs to match against.
-
-        Returns:
-            A list of indexes for matching expectation objects.
-            If there are no matches, the list will be empty.
+    def remove_expectation(
```

Comment:
rename: `remove_expectations`

Reply:
1. Add guard `remove_multiple` with default false.
2. Add utility to remove all expectations of a given type which loops over domain kwargs for expectation (e.g.).